File: 3251-maximum-area-of-longest-diagonal-rectangle/3251-maximum-area-of-longest-diagonal-rectangle.c
int areaOfMaxDiagonal(int** dimensions, int dimensionsSize, int* dimensionsColSize) {
    double max_diagonal = 0;
    int answer = 0;

    for (int i=0; i<dimensionsSize; i++) {
        int length = dimensions[i][0];
        int width = dimensions[i][1];
        double diag = sqrt((pow(length, 2) + pow(width, 2)));
        
        if (diag > max_diagonal) {
            max_diagonal = diag;
            answer = length * width;
        } else if (diag == max_diagonal) {
            if (length * width > answer) {
                answer = length * width;
            }
        }
    }

    return answer;
}

File: 3251-maximum-area-of-longest-diagonal-rectangle/3251-maximum-area-of-longest-diagonal-rectangle.py
class Solution:
    def areaOfMaxDiagonal(self, dimensions: List[List[int]]) -> int:
        max_diagonal = 0
        answer = 0
        for length, width in dimensions:
            if (length ** 2 + width ** 2) ** (1/2) > max_diagonal:
                max_diagonal = (length ** 2 + width ** 2) ** (1/2)
                answer = length * width
            elif (length ** 2 + width ** 2) ** (1/2) == max_diagonal:
                answer = max(length * width, answer)

        return answer